website/static/js/contribRemover.js

```diff
@@ -0,0 +1,275 @@
+/**
+ * Controller for the Remove Contributor modal.
+ */
+'use strict';
+
+var $ = require('jquery');
+var ko = require('knockout');
+var bootbox = require('bootbox');
+var Raven = require('raven-js');
+
+var oop = require('./oop');
+var $osf = require('./osfHelpers');
+var Paginator = require('./paginator');
+
+var RemoveContributorViewModel = oop.extend(Paginator, {
+    constructor: function(title, nodeId, userName, userId, contribShouter, pageChangedShouter) {
+        this.super.constructor.call(this);
+        var self = this;
+        self.title = title;
+        self.nodeId = nodeId;
+        self.userId = userId;
+        self.contributorToRemove = ko.observable('');
+        self.REMOVE = 'remove';
+        self.REMOVE_ALL = 'removeAll';
+        self.REMOVE_NO_CHILDREN = 'removeNoChildren';
+        self.REMOVE_SELF = 'removeSelf';
+
+        //This shouter allows the ContributorsViewModel to share which contributor to remove
+        // with the RemoveContributorViewModel
+        contribShouter.subscribe(function(newValue) {
+            self.contributorToRemove(newValue);
+        }, self, 'contribMessageToPublish');
+
+        //This shouter allows RemoveContributorViewModel to know if the
+        // ContributorsViewModel is in a dirty state to prevent removal
+        self.pageChanged = ko.observable(false);
+        pageChangedShouter.subscribe(function(newValue) {
+            self.pageChanged(newValue);
+        }, self, 'changedMessageToPublish');
+
+        self.page = ko.observable(self.REMOVE);
+        self.pageTitle = ko.computed(function() {
+            return {
+                remove: 'Delete Contributor',
+                removeAll: 'Delete Contributor',
+                removeNoChildren: 'Delete Contributor'
+            }[self.page()];
+        });
+        self.userName = ko.observable(userName);
+        self.deleteAll = ko.observable(false);
+        var nodesOriginal = {};
+        self.nodesOriginal = ko.observableArray();
```

Comment:
`self.nodesOriginal` is treated as an Object in the code, but you are instantiating it as an `observableArray`, which is confusing. 

Does it need to be an observable at all?